packages/slate-editor/src/modules/rich-formatting-menu/RichFormattingMenu.tsx

```diff
@@ -109,14 +109,24 @@ export function RichFormattingMenu({
     function handleLinkButtonClick() {
         if (!editor.selection) return;
 
-        Transforms.select(editor, editor.selection.anchor.path);
-
         const rangeRef = Editor.rangeRef(editor, editor.selection, {
             affinity: 'inward',
         });
 
         setLinkRange(rangeRef);
 
+        if (rangeRef.current) {
+            const hasLink = getCurrentLinkNode(editor, { at: rangeRef.current });
+            const selectedOnlyLink = Path.equals(
+                rangeRef.current.anchor.path,
+                rangeRef.current.focus.path,
+            );
+
+            if (hasLink && selectedOnlyLink) {
+                Transforms.select(editor, rangeRef.current.anchor.path);
+            }
+        }
+
```

Comment:
That was not possible before
This section was added in case if you want to change url, otherwise you can make a mistake in selection. Select part of the link and change the url,  in that case you would have two link with different url.

Reply:
> That was not possible before

Yes. That's a bug :) 

> This section was added in case if you want to change url, otherwise you can make a mistake in selection. Select part of the link and change the url, in that case you would have two link with different url.

Protecting against mistakes is a good thing. But here we make a certain use-case impossible. So I think it's not worth it.

Reply:
Removed this code, however unlink part of the link still not possible because `unwrapLink` remove link signature from whole node